### PR TITLE
feat(growth): add abstract onboarding model

### PR DESCRIPTION
**Pull request by roggenkemper**
this pr adds an abstract onboarding task model that can be subclassed. the original OrganizationOnboardingTask model is unchanged from this change, but this change allows for the Sandbox to create a subclass that allows the Tasks to be unique based on user, instead of just being organization based. 

GRW-633

### CODE REVIEW

File: src/sentry/models/organizationonboardingtask.py [hunk 1]
```diff
@@ -71,14 +71,46 @@ def record(self, organization_id, task, **kwargs):
...
+    date_completed = models.DateTimeField(default=timezone.now)
+    project = FlexibleForeignKey("sentry.Project", db_constraint=False, null=True)
+    data = JSONField()  # INVITE_MEMBER { invited_member: user.id }
+
+    objects = OrganizationOnboardingTaskManager()
+
+    class Meta:
+        abstract = True
+
+    __repr__ = sane_repr("organization", "task")
```
Comment (scefali) [inline src/sentry/models/organizationonboardingtask.py:105]: @roggenkemper I think this needs to move out of the abstract class and the sandbox version need `user` in there

File: src/sentry/models/organizationonboardingtask.py [hunk 2]
```diff
@@ -71,14 +71,46 @@ def record(self, organization_id, task, **kwargs):
...
+    user = FlexibleForeignKey(
+        settings.AUTH_USER_MODEL, null=True, on_delete=models.SET_NULL
+    )  # user that completed
+    status = BoundedPositiveIntegerField(choices=[(k, str(v)) for k, v in STATUS_CHOICES])
+    completion_seen = models.DateTimeField(null=True)
+    date_completed = models.DateTimeField(default=timezone.now)
+    project = FlexibleForeignKey("sentry.Project", db_constraint=False, null=True)
+    data = JSONField()  # INVITE_MEMBER { invited_member: user.id }
+
+    objects = OrganizationOnboardingTaskManager()
```
Comment (scefali) [inline src/sentry/models/organizationonboardingtask.py:100]: @roggenkemper this should also be moved out of the abstract class

File: src/sentry/models/organizationonboardingtask.py [hunk 3]
```diff
@@ -71,14 +71,46 @@ def record(self, organization_id, task, **kwargs):
         return False
 
 
-@region_silo_only_model
-class OrganizationOnboardingTask(Model):
+class AbstractOnboardingTask(Model):
```
Comment (scefali) [inline src/sentry/models/organizationonboardingtask.py:74]: @roggenkemper can you also add a comment explaining we use in the sandbox? We should document all these kind of imports so someone doesn't later come in and wonder why this abstract class exists.

File: src/sentry/models/organizationonboardingtask.py (hunk 1 above)
Comment (roggenkemper) [reply src/sentry/models/organizationonboardingtask.py:105]: yup, the sandbox version has user here but ill remove this from the abstract version

**Review by scefali — APPROVED** [review]

**Pull request merged**